### PR TITLE
Rework rssi code a bit

### LINKED ISSUE #5500: RSSI Shutdown timeout on X9D+/SHUTDOWN_CONFIRMATION on other platforms
SHUTDOWN_CONFIRMATION disables RSSI check on shutdown via #idef, this seem wrong and counter intuative.

On the X9D+ if you don't switch power back on and just let the screen sit there, the TX turn off after almost exactly 10s. If you compile the X9D+ with PWR_BUTTON_PRESS the timeout disappear.

### PR DESCRIPTION
**Pull request by schwabe**
Fixes #5500

### CODE REVIEW

File: radio/src/targets/taranis/CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +33,7 @@ elseif(PCB STREQUAL X9D+)
   set(HAPTIC YES)
   set(LUA_EXPORT lua_export_x9d)
   set(FLAVOUR x9d+)
+  set (PWR_BUTTON_DEFAULT "SWITCH")
```
Comment (bsongis) [inline radio/src/targets/taranis/CMakeLists.txt]: It's not possible to have PRESS here

File: radio/src/targets/taranis/CMakeLists.txt [hunk 2]
```diff
@@ -56,7 +56,6 @@ elseif(PCB STREQUAL X9D)
   set(SERIAL2_DRIVER ../common/arm/stm32/serial2_driver.cpp)
   set(GVAR_SCREEN model_gvars.cpp)
 elseif(PCB STREQUAL X7)
-  set(PWR_BUTTON "PRESS" CACHE STRING "Pwr button type (PRESS/SWITCH)")
   set(CPU_TYPE STM32F2)
```
Comment (bsongis) [inline radio/src/targets/taranis/CMakeLists.txt]: the definition below should be removed as well

**Comment by bsongis** [comment]
I will revert the CMakeLists.txt for merging it quickly

**Comment by bsongis** [comment]
Tested on X9D+

File: radio/src/targets/taranis/CMakeLists.txt (hunk 1 above)
Comment (schwabe) [reply radio/src/targets/taranis/CMakeLists.txt]: That only sets the default the real setting is further down

**Pull request merged**